docs: Add documentation review guidance (#3495)
NVIDIA/bare-metal-manager-core: AGENTS.md

exposed name, verbatim. Existing metric names never change. The full standard
lives in [`docs/observability/instrumentation.md`](docs/observability/instrumentation.md).

## Documentation review

These are documentation checks, not style guidance. Apply every relevant
check before requesting review.

- **Interface contracts:** Document the complete contract, not just the name or happy path.

  - For every documented command, flag, environment variable, config key, API
    field, mode, or state, verify from code, schema, or exercised output: exact
    spelling and case; required or optional condition; default; accepted
    values, units, formats, and bounds; mutual exclusions and interactions;
    global versus subcommand position and required order; omission or fallback
    behavior; and observable output, side effects, errors, and unsupported

>     paths.

Reviewer: Let's simplify this formatting a bit for clarity

```suggestion
  - For every documented command, flag, environment variable, config key, API field, mode, or state, verify the following from code, schema, or exercised output:
    - exact spelling and case
    - required or optional condition
    - default
    - accepted values, units, formats, and bounds
    - mutual exclusions and interactions
    - global versus subcommand position and required order
    - omission or fallback behavior
    - observable output, side effects, errors, and unsupported paths
```

  - Exercise each changed example at the PR revision on an authorized local or
    test target and compare it with real `--help` output.
  - If any answer is unknown, stop and ask the owner; another documentation page
    is not evidence.

- **Generated interfaces:** Change the source, regenerate every output, and prove they stay in sync.

>   - Never edit a generated reference alone.

Reviewer: Is "alone" useful here? Seems like the guidance should actually be _just don't edit anything generated_.
```suggestion
  - Never edit a generated reference.
```

  - For `nico-admin-cli`, change the Clap declarations under
    `crates/admin-cli/src/`, verify the affected command with
    `cargo run -q -p nico-admin-cli -- <command-path> --help`, then run
    `cargo make gen-cli-docs` and `cargo make check-cli-docs`. See
    `crates/admin-cli/AGENTS.md` for the generated and hand-authored boundaries.
  - For REST, use `rest-api/openapi/spec.yaml` for the contract and inspect the
    handler or model for conditional behavior the schema cannot express. When
    the spec changes, run `make rest-api/lint-openapi`,
    `make rest-api/generate-sdk`, `make rest-api/publish-openapi`, and
    `make openapi-breaking`; do not edit `rest-api/sdk/standard/` or

>     `rest-api/docs/index.html` alone.

Reviewer: Again, strike "alone" here?
```suggestion
    `rest-api/docs/index.html`.
```

- **Workflow parity:** Make the documentation match the workflow that actually runs.

  - For setup documentation, `helm-prereqs/setup.sh` is the source of truth for
    phases, skip flags, environment requirements, and component order. Run
    `bash -n helm-prereqs/setup.sh` and cross-check `helm-prereqs/README.md` and
    `docs/getting-started/quick-start.md`.
  - For state-machine documentation, trace every success, skip, retry, poll,
    restart, deletion, maintenance, and error transition to its enum and
    handler. Narrative, Mermaid, and transition tables must contain the same
    states and edges, including persisted resume state.

- **Metric catalogue:** A metric is not documented until its HELP text, emitted series, and generated catalogue agree.

>   - Treat metric HELP text and `docs/observability/core_metrics.md` as generated

Reviewer: But isn't the HELP text human-written?

Author: It is but I think its pulled from the openapi.yaml spec and placed into the cli help text

    API documentation.
  - Verify what causes the observation, its counter/gauge/histogram type, the
    exact entity and condition measured, direction or protocol, and every label
    dimension.
  - New metrics need non-empty `describe` text and must be exercised by
    `test_integration` so catalogue generation includes their exposed name,

>     type, and description. Do not patch the generated table alone.

Reviewer: Again, "alone" here? Don't we want to say "Do not patch the generated table", period?

> - **Cross-surface drift:** Change a fact everywhere it appears or make one page canonical and link the rest.

Reviewer: ```suggestion
- **Cross-surface drift:** Change a fact everywhere it appears or make one page canonical and link to the canonical page from the others.
```

  - Search every changed literal or behavior with
    `rg -n --fixed-strings '<literal>' README.md crates/ rest-api/ docs/ book/ helm/ helm-prereqs/ deploy/`;
    reconcile every conflicting hit or establish one canonical explanation and
    link the others to it.
  - New or moved public pages must be present in `docs/index.yml`, and changed

>     public paths need redirects in `fern/docs.yml`.

Reviewer: Running `fern docs md check` and `fern check` can surface issues with moved pages, and suggest redirects. Not sure every agent will have a Fern token, though.

  - `docs/release-notes.md` owns current unified releases;
    `rest-api/CHANGELOG.md` is legacy history whose published entry order must

>     be preserved.

Reviewer: ```suggestion
    do not modify `rest-api/CHANGELOG.md`, as it is legacy history whose published entry order must be preserved.
```

- **Temporary claims and versions:** Tie temporary claims and pinned versions to a real support boundary.

  - Search changed prose for `currently`, `today`, `for now`, `temporarily`, and
    `draft`. Each match must name a release, deliberate support boundary, or
    full tracking URL; a bare issue number or review-history note is
    insufficient.
  - Hard-code a tool or dependency version only when it is a tested minimum,
    maximum, or pinned compatibility boundary; otherwise link the authoritative
    release page.

- **Rendered output:** Review each artifact in the renderer its readers actually use.

  - Run `rumdl check --config docs/.rumdl.toml <changed-markdown-files>` and fail
    on every finding.
  - For Fern-published pages, run `fern check` and inspect the PR preview; if it

>     is unavailable, use `(cd fern && fern docs dev)`.

Reviewer: Fern check doesn't generate a preview. If you want to include preview-generation steps, do it with an `--id` flag, and make sure to delete the preview when finished.
